service/src/models/archaeologist.ts

```diff
@@ -67,6 +69,35 @@ export class Archaeologist {
       } catch (err) {
         console.log("problem with pipe", err)
       }
+    });
+
+    this.node.handle(['/validate-arweave'], async ({ stream }) => {
+      try {
+        await pipe(stream, async (source) => {
+          for await (const data of source) {
+            const jsonData = JSON.parse(new TextDecoder().decode(data));
+
+            const txId = jsonData.arweaveTxId;
+            const unencryptedShardHash = jsonData.unencryptedShardHash;
+
+            const wallet = new ethers.Wallet(process.env.ENCRYPTION_PRIVATE_KEY!);
+
+            const isValidShard = await fetchAndValidateArweaveShard(txId, unencryptedShardHash, wallet.publicKey);
+
+            if (isValidShard) {
+              const msg = ethers.utils.solidityPack(['string', 'string'], [txId, unencryptedShardHash])
+              const signature = await wallet.signMessage(msg);
+
+              const toBrowser = JSON.stringify(signature);
+              this.publish("env-config", `signature ${toBrowser}`);
```

Comment:
Once streaming is setup, this could be piped through on a `signature` protocol. The web app will be receiving multiples of these back, and will need to know which archaeologist they came from so it (the web app) can validate the signatures.